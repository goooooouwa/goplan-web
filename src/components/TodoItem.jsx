import { ListItem, ListItemButton, ListItemText } from "@mui/material";
import React from "react";
import { Link } from "react-router-dom";

export default function TodoItem(props) {
  return (
    <>
      <ListItem disablePadding>
        <ListItemButton component={Link} to={`/projects/${props.todo.projectId}/todos/${props.todo.id}`} >
          <ListItemText primary={props.todo.name} />
        </ListItemButton>
      </ListItem>
    </>
  );
}
