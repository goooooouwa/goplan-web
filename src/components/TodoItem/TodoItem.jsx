import { Checkbox, ListItem, ListItemButton, ListItemText } from "@mui/material";
import React from "react";
import PropTypes from 'prop-types';
import { Link } from "react-router-dom";

export default function TodoItem(props) {
  return (
    <>
      <ListItem disablePadding>
        <Checkbox
          checked={props.todo.status}
          onChange={(event)=>{props.handleTodoChange(event, props.todo)}}
          inputProps={{ 'aria-label': 'controlled' }}
        />
        <ListItemButton component={Link} to={`/projects/${props.todo.projectId}/todos/${props.todo.id}`} >
          <ListItemText primary={props.todo.name} />
        </ListItemButton>
      </ListItem>
    </>
  );
}

TodoItem.propTypes = {
  todo: PropTypes.shape({
    id: PropTypes.number,
    projectId: PropTypes.number,
    status: PropTypes.bool,
    name: PropTypes.string
  }).isRequired,
  handleTodoChange: PropTypes.func.isRequired
};