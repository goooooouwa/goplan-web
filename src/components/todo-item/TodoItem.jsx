import React from "react";

export default function TodoItem(props) {
    return <li>
        {props.todo.name} |{' '}
        {props.todo.description} |{' '}
        {props.todo.timeSpan} |{' '}
        {props.todo.startDate} |{' '}
        {props.todo.endDate} |{' '}
        {props.todo.repeat} |{' '}
        {props.todo.repeatPeriod} |{' '}
        {props.todo.repeatTimes} |{' '}
        {props.todo.instanceTimeSpan} |{' '}
        {props.todo.dependencies} |{' '}
        {props.todo.dependents} |{' '}
    </li>;
}
