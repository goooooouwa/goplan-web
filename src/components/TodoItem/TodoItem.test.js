import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, screen, render } from '@testing-library/react';
import TodoItem from './TodoItem';
import { BrowserRouter } from 'react-router-dom';

const todo = {
  id: 1,
  projectId: 1,
  status: false,
  name: "Todo 1"
};

it('should show a todo item', () => {
  const component = renderer.create(
    <BrowserRouter>
      <TodoItem
        todo={todo}
        handleTodoChange={() => { }}
      />
    </BrowserRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should display todo name', () => {
  render(
    <BrowserRouter>
      <TodoItem
        todo={todo}
        handleTodoChange={() => { }}
      />
    </BrowserRouter>
  );

  expect(screen.getByText(/Todo 1/)).toBeInTheDocument();
  expect(screen.getByRole('checkbox')).toBeInTheDocument();

  // expect(screen.getByRole('checkbox').checked).toBeFalsy();

  // fireEvent.click(screen.getByRole('checkbox'));

  // expect(await screen.getByRole('checkbox').checked).toBeTruthy();
});