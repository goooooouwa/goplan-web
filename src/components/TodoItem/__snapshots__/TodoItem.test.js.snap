// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show a todo item 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
  disabled={false}
>
  <span
    className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      aria-label="controlled"
      checked={false}
      className="PrivateSwitchBase-input css-1m9pwf3"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="CheckBoxOutlineBlankIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
  <a
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
    href="/projects/1/todos/1"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Todo 1
      </span>
    </div>
  </a>
</li>
`;
